### PR TITLE
set resource requests

### PR DESCRIPTION
**Pull request by ashleyschuett**
**What this PR does / why we need it**:

Add resource requests to `virt-controller`, `virt-api`, `virt-operator` and `virt-handler`. You can look at this change in the first commit in the second commit I moved DS to it's own file since the naming did not make sense. We should have resource requests set on these resources to allow the Kubernetes schedule to make smarter decisions during scheduling and when OOM killing pods. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Add resource requests to virt-controller, virt-api, virt-operator and virt-handler
```

### CODE REVIEW

File: pkg/virt-operator/resource/generate/components/daemonsets.go [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+					Path: volume.path,
+				},
+			},
+		})
+	}
+
+	container.Resources = corev1.ResourceRequirements{
+		Requests: corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("10m"),
+			corev1.ResourceMemory: resource.MustParse("200Mi"),
```
Comment (stu-gott) [inline pkg/virt-operator/resource/generate/components/daemonsets.go:233]: I've seen virt-handler consume as much as 183Mi in the wild. I would imagine we don't need to cover all cases, but 200Mi might be cutting it a tiny bit close.

**Comment by stu-gott** [comment]
/approve

Sans the question about virt-handler's resource consumption, this PR looks in good shape.

File: pkg/virt-operator/resource/generate/components/daemonsets.go (hunk 1 above)
Comment (ashleyschuett) [reply pkg/virt-operator/resource/generate/components/daemonsets.go]: I've added ~50Mi padding so 230Mi total. Does that seem reasonable? Or should there be more?

**Comment by yangxiaoliang** [comment]
Hi @ashleyschuett  @stu-gott , I have a question on this PR, shall we need to add resource limit to virt-controller, virt-api, virt-operator and virt-handler ?

**Comment by rmohr** [comment]
> Hi @ashleyschuett @stu-gott , I have a question on this PR, shall we need to add resource limit to virt-controller, virt-api, virt-operator and virt-handler ?

Hey, setting resource limits is something one should avoid. The issue with limits is that one gets OOM killed if one gets above resources and the actual memory usage depends on many things, like kernel version, node architecture and so forth. As a consequence we would have to set them VERY high to be on the safe side. Our goal should be to aim for requests where we are almost all the time below the requested value.

**Comment by yangxiaoliang** [comment]
> > Hi @ashleyschuett @stu-gott , I have a question on this PR, shall we need to add resource limit to virt-controller, virt-api, virt-operator and virt-handler ?
> 
> Hey, setting resource limits is something one should avoid. The issue with limits is that one gets OOM killed if one gets above resources and the actual memory usage depends on many things, like kernel version, node architecture and so forth. As a consequence we would have to set them VERY high to be on the safe side. Our goal should be to aim for requests where we are almost all the time below the requested value.

Thanks for your reply @rmohr, but add a resource limits can avoid virt-controller/virt-api/virt-operator/virt-handler takes
 additional resources, in worst cases, avoid the host resources are exhausted by them, what's your opinion here ?

**Pull request merged**